packages/shared/src/variants/quantum.ts

```diff
@@ -234,8 +250,53 @@ export class QuantumGo extends AbstractGame<NewBadukConfig, QuantumGoState> {
     }
     throw new Error("Error finding mapped capture.");
   }
+
+
+  public bootstrap(playerB: string, playerW: string, boardSize: number, komi: number){
+
+    const currentDate: Date = new Date();
+
+    const initData: string[] = [
+
+      "(;\n", 
+      "EV[GO Variants]\n", 
+      `PB[${playerB}]\n`, 
+      `PW[${playerW}]\n`,
+      `SZ[${boardSize}]\n`,
+      `KM[${komi}]\n`,     
+      `DT[${currentDate.getMonth()+1}/${currentDate.getDate()}/${currentDate.getFullYear()} ${currentDate.getHours()}:${currentDate.getMinutes()}]\n`,
+      "VS[quantum]\n",
+      "\n",
+      "\n"
+
+    ];
+  
+    this.writeToSGF(initData, true);
+   
+  }
+  
+  //creating sgf file
+  public writeToSGF(text: string[], callingBootstrap: boolean){
+
+    for (let i = 0; i < text.length; i++){
+      let formattedMove = text[i];
+      if(!callingBootstrap){
+        formattedMove = `${i % 2 === 0 ? ";B" : ";W"}[${text[i]}]`;
+      }
+      this.sgfContent += formattedMove;
+    }
+    if(!callingBootstrap){
+      this.sgfContent += "\n\n)";
+    }  
+  }
+
+  getSGF(): string{
```

Comment:
Somewhere we should check whether this game is played on a grid board or not, because non-rectangular boards are unfortunately not compatible with SGF.

For example:

```
getSGF(): string {
  if (!isGridBadukConfig(this.config)) {
    // SGF for non rectangular boards is not possible
    return "";
  }
  return this.sgfContent;
}
```